src/proto/streams/state.rs

```diff
@@ -413,11 +413,8 @@ impl State {
         matches!(self.inner, Closed(_))
     }
 
-    pub fn is_recv_closed(&self) -> bool {
-        matches!(
-            self.inner,
-            Closed(..) | HalfClosedRemote(..) | ReservedLocal
```

Comment:
I've given this some thought, and I'm less sure that we want the check to be so strict. If it is half-closed successfully, I think we _do_ want to say that the `RecvStream` is finished. If there is later an error on the sending side, I don't think that should be a problem, the `SendStream` will notice.

Reply:
Good point. Updated to check both `Closed(Cause::EndStream)` and  `HalfClosedRemote(..)`. Also updated the function name accordingly.